### ISSUE TITLE
Script forces old deprecated MSonline version

### ISSUE THREAD

**Issue by ryanedlund** [Azure-Samples/azure-mfa-authentication-method-analysis#8]
Can this script be updated to support the new Microsoft.Graph module instead of MSOnline [1.1.183.57]?

**Comment by marcgarciamarti** [comment]
Let me confirm I understand this correctly, because I can't get this to work. 

This script is linked from the official documentation page: https://learn.microsoft.com/en-us/entra/identity/authentication/howto-mfa-getstarted#choose-authentication-methods-for-mfa but it relies on a phased out stack?

thanks

**Comment by ryanedlund** [comment]
> Let me confirm I understand this correctly, because I can't get this to work.
> 
> This script is linked from the official documentation page: https://learn.microsoft.com/en-us/entra/identity/authentication/howto-mfa-getstarted#choose-authentication-methods-for-mfa but it relies on a phased out stack?
> 
> thanks

Yes, I believe they now have a native way to do this with the Graph module instead:

https://learn.microsoft.com/en-us/powershell/module/microsoft.graph.reports/get-mgreportauthenticationmethoduserregistrationdetail?view=graph-powershell-1.0
